st2client/st2client/utils/types.py

```diff
@@ -0,0 +1,84 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Based on http://code.activestate.com/recipes/576694/ (MIT license)
+"""
+
+from __future__ import absolute_import
+import collections
+
+__all__ = [
+    'OrderedSet'
+]
+
+
+class OrderedSet(collections.MutableSet):
```

Comment:
Wish there were associated tests for this.

Reply:
Will add them.

In fact, that's copied from st2common (because of our unresolved dependencies between st2common and st2client).

Reply:
Here we go - a387c03.